src/proto/console/text/output.rs

```diff
@@ -30,37 +30,37 @@ pub struct Output<'boot> {
 
 impl<'boot> Output<'boot> {
     /// Resets and clears the text output device hardware.
-    pub fn reset(&mut self, extended: bool) -> Result<()> {
+    pub fn reset(&mut self, extended: bool) -> Result {
         (self.reset)(self, extended).into()
     }
 
     /// Clears the output screen.
     ///
     /// The background is set to the current background color.
     /// The cursor is moved to (0, 0).
-    pub fn clear(&mut self) -> Result<()> {
+    pub fn clear(&mut self) -> Result {
         (self.clear_screen)(self).into()
     }
 
     /// Writes a string to the output device.
-    pub fn output_string(&mut self, string: &CStr16) -> Result<()> {
+    pub fn output_string(&mut self, string: &CStr16) -> Result {
         unsafe { (self.output_string)(self, string.as_ptr()) }.into()
     }
 
     /// Checks if a string contains only supported characters.
     ///
     /// UEFI applications are encouraged to try to print a string even if it contains
     /// some unsupported characters.
-    pub fn test_string(&mut self, string: &CStr16) -> bool {
+    pub fn test_string(&mut self, string: &CStr16) -> Result<bool> {
```

Comment:
Thanks to UEFI's extensible error handling design, this function could, in principle at least, fail to answer the question that we asked.